tests/background/tab-state-test.js

```diff
@@ -132,23 +132,126 @@ describe('TabState', () => {
   });
 
   describe('#updateAnnotationCount', () => {
-    afterEach(() => {
-      $imports.$restore();
-    });
+    let clock;
+    let getStub;
 
-    it('queries the service and sets the annotation count', () => {
-      const testValue = 42;
-      const getStub = sinon.stub().returns(Promise.resolve(testValue));
+    beforeEach(() => {
+      clock = sinon.useFakeTimers();
+      getStub = sinon.stub();
       $imports.$mock({
         './uri-info': {
           getAnnotationCount: getStub,
         },
       });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+      clock.restore();
+    });
+
+    it('queries the service and sets the annotation count after waiting for a period of 3000ms', async () => {
+      const testValue = 42;
+      getStub.resolves(testValue);
+      const tabState = new TabState({ 1: { state: states.ACTIVE } });
+
+      const promise = tabState.updateAnnotationCount(1, 'foobar.com');
+      clock.tick(1010);
+      await promise;
+      assert.called(getStub);
+      assert.equal(tabState.getState(1).annotationCount, testValue);
+    });
+
+    it('cancels the first query (during waiting period) when the service is called two consecutive times for the same tab', async () => {
+      const testValue = 42;
+      getStub.resolves(testValue);
+      const tabState = new TabState({ 1: { state: states.ACTIVE } });
+
+      const promise1 = tabState.updateAnnotationCount(1, 'foobar.com');
+      clock.tick(800); // promise 1 is still waiting when promise2 is called
+      const promise2 = tabState.updateAnnotationCount(1, 'foobar.com');
+      clock.tick(2300);
+      await promise1;
+      assert.strictEqual(tabState.getState(1).annotationCount, 0);
```

Comment:
The code uses a mix of `assert.equal(...)` and `assert.strictEqual(...)`. We should be consistent about using one or the other in these tests. Either would be fine. We tend to use `assert.equal(...)` most of the time, although `assert.strictEqual(...)` can in principle catch certain additional mistakes.